### PR TITLE
More compact code.

### PR DESCRIPTION
**Pull request by mbebenita**
Depends on PR #1203

### CODE REVIEW

File: jit/baseline.ts [hunk 1]
```diff
@@ -312,6 +312,8 @@ module J2ME {
      */
     static stackNames = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "_O", "P", "Q", "R", "S", "T", "_U", "V", "W", "X", "Y", "Z"];
 
+    private hasUnwindThrow;
```
Comment (yxl) [inline jit/baseline.ts]: nit: `private hasUnwindThrow: boolean;`

File: jit/baseline.ts [hunk 2]
```diff
@@ -556,18 +564,20 @@ module J2ME {
 
       if (needsOSREntryPoint) {
         // Are we doing an OSR?
-        this.bodyEmitter.enter("if (O) {");
-        this.bodyEmitter.writeLn("var local = O.local;");
+        this.bodyEmitter.enter("if(O){");
+        this.bodyEmitter.writeLn("var _ = O.local;");
```
Comment (yxl) [inline jit/baseline.ts]: "var _=O.local;"

File: jit/baseline.ts [hunk 3]
```diff
@@ -995,13 +1012,13 @@ module J2ME {
       if (!emitCheckPreemption) {
         return;
       }
-      emitter.writeLn("PS ++;");
-      emitter.writeLn("if ((PS & " + preemptionSampleMask + ") === 0) PE();");
+      emitter.writeLn("PS++;");
+      emitter.writeLn("if((PS & " + preemptionSampleMask + ")===0)PE();");
```
Comment (yxl) [inline jit/baseline.ts]: Two more spaces could be removed: `if((PS&`

**Comment by yxl** [comment]
I reviewed the first three commits regarding removing the white spaces. The last two commits are beyond my knowledge and need someone else to review :-)
1. 46b9fd2: A more compact encoding of unwind code.
2. 85e33fd: Comments and assertions.

**Comment by mbebenita** [comment]
@yxl thanks for the review.

**Comment by marco-c** [comment]
Myk reviewed 46b9fd2 and 85e33fd in #1203.
I've reviewed the last few commits.

**Pull request merged**